### PR TITLE
Fix wrong back navigation transition animation in transfer balance page

### PR DESCRIPTION
**Pull request by bernhardoj**
### Details

The back navigation in transfer balance page slides the page from right to left.

### Fixed Issues

$ https://github.com/Expensify/App/issues/37982
PROPOSAL: https://github.com/Expensify/App/issues/37982#issuecomment-1986053364

### Tests

Same as QA Steps
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as QA Steps
### QA Steps

1. Go to account settings > wallet > transfer balance
2. Click the header back button
3. Verify the transfer balance page close animation is slides to the right
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/50919443/cdf37649-0d08-4209-9bc1-1acc7da9fa16

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/50919443/06a3f989-8d3e-46b8-94e1-dfeec65d2d3b

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/50919443/863d1b62-c0e5-416f-8580-315b2c33d152

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/50919443/2576c9a3-15c3-44ee-af07-0e203326d9b0

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/50919443/8b6ff629-234e-41f5-882c-3b4d795c0cf3

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/50919443/bc4b3df4-0932-4c2a-bed4-af566476768c

</details>

### CODE REVIEW

File: src/pages/settings/Wallet/TransferBalancePage.tsx [hunk 1]
```diff
@@ -156,12 +156,10 @@ function TransferBalancePage({bankAccountList, fundList, userWallet, walletTrans
                 titleKey="notFound.pageNotFound"
                 subtitleKey="transferAmountPage.notHereSubTitle"
                 linkKey="transferAmountPage.goToWallet"
-                onLinkPress={() => Navigation.goBack(ROUTES.SETTINGS_WALLET)}
             >
                 <HeaderWithBackButton
                     title={translate('common.transferBalance')}
                     shouldShowBackButton
-                    onBackButtonPress={() => Navigation.goBack(ROUTES.SETTINGS_WALLET)}
```
Comment (bernhardoj) [inline src/pages/settings/Wallet/TransferBalancePage.tsx:164]: The default props is Navigation.goBack

File: src/pages/settings/Wallet/TransferBalancePage.tsx [hunk 2]
```diff
@@ -156,12 +156,10 @@ function TransferBalancePage({bankAccountList, fundList, userWallet, walletTrans
                 titleKey="notFound.pageNotFound"
                 subtitleKey="transferAmountPage.notHereSubTitle"
                 linkKey="transferAmountPage.goToWallet"
-                onLinkPress={() => Navigation.goBack(ROUTES.SETTINGS_WALLET)}
```
Comment (bernhardoj) [inline src/pages/settings/Wallet/TransferBalancePage.tsx:159]: To cover the text button of the not found page case. The default props is Navigation.dismissModal

https://github.com/Expensify/App/assets/50919443/e1a9d979-1740-4081-b890-4e68092ee710

**Comment by mountiny** [comment]
@bernhardoj @jayeshmangwani can you test deeplinking to the transfer balance page and then going back?

**Comment by bernhardoj** [comment]
Sure.

Works fine on web

https://github.com/Expensify/App/assets/50919443/ad6744b9-fbed-435d-bb41-6c31c32cfdc1

https://github.com/Expensify/App/assets/50919443/f03eb873-511f-45f4-806c-4b782b5d10ec

But it looks weird on the native

https://github.com/Expensify/App/assets/50919443/e70cba81-93f1-41db-b4e7-6119a1d678c6

1. When I deep link to the transfer balance page, the LHN slides to the right.
2. The page is pushed twice to the stack, so pressing back will show the same page. It's caused by this [code](https://github.com/Expensify/App/blob/9302ade9988b1cba3e07b48f474bc35a6deea7fb/src/libs/actions/Report.ts#L2271)

It's unrelated to this PR though as this happens on other pages too that are 2 levels deep in the settings page.

https://github.com/Expensify/App/assets/50919443/2cb95525-eff4-42d5-aa09-da0204b07c49

**Comment by jayeshmangwani** [comment]
@bernhardoj Can you please include the steps, how can you able to access transfer balance page ?
I have tried to set userWallet to GOLD in dev, but still I am not able to access the page and navigating to to debit card page.

**Comment by jayeshmangwani** [comment]
https://github.com/Expensify/App/assets/35371050/40086b8f-9dcc-41f1-bdf1-56e109ddda9f

@bernhardoj Need to fix transfer successful animation too

**Comment by jayeshmangwani** [comment]
> can you test deeplinking to the transfer balance page and then going back?

@mountiny I am now able to test Transfer balance by setting the the wallet to GOLD in dev environment

mWeb and web is working fine
<details>
  <summary>mWeb , web</summary>
 

https://github.com/Expensify/App/assets/35371050/6d7da3f7-5475-4791-9af4-c10074699ecf


https://github.com/Expensify/App/assets/35371050/d96728a8-2b97-4c8c-91df-f8a10d7046ea


</details>


But getting an issue in the iOS native device when using the deeplink, when pressing the back icon it goes to the not found page and need to press back again to navigate back to the wallet page

<details>
  <summary>iOS</summary>

https://github.com/Expensify/App/assets/35371050/850c8e04-2385-495a-89e4-9aa03f86d3ad


</details>

And i think that issue is not related to this PR, if we revert this PR then also issue is happening in the main branch, and also I found similar issue with other pages, e.g. Status page

<details>
  <summary>Status Page</summary>


https://github.com/Expensify/App/assets/35371050/cfd12504-6ef3-488a-8f44-3a09827929cb



</details>

**Comment by bernhardoj** [comment]
>  Can you please include the steps, how can you able to access transfer balance page ?

I change the code to access the page 😅 

> Need to fix transfer successful animation too

Fixed.

https://github.com/Expensify/App/assets/50919443/a326c8a4-b4b7-4995-8d0a-b6c7ba4f612e

**Comment by jayeshmangwani** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/35371050/232a9279-71ed-4c39-bd2e-677304f213d5




</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/35371050/c932f58a-4747-47e0-bbd5-563bd01efa20




</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/35371050/c855ed63-06de-475e-aad7-3e7d266ff788




</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/35371050/dd72e99f-8098-428f-b83a-a0fe0671d4c4




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/35371050/12706352-3f25-4c18-8074-81d504859db3





</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/35371050/8872d84e-3bc8-4daf-8e42-ab0cef3ad992




</details>

**Comment by jayeshmangwani** [comment]
LGTM

**Review by jayeshmangwani — APPROVED** [review]

**Review by mountiny — APPROVED** [review]
Thanks

**Comment by kavimuru** [comment]
@bernhardoj "Transfer balance" button should be only visible when user has some balance in wallet, but Applause don't have it.
Could you validate internally?

**Comment by bernhardoj** [comment]
@kavimuru I think you need a wallet with a Gold/Platinum tier.

@jayeshmangwani can you share how do you change the wallet to Gold?

**Comment by jayeshmangwani** [comment]
@kavimuru @bernhardoj I was able to test by calling the Onyx mock data on wallet page locally, let me give you mock data and attaching a video how I had activated locally transfer balance, took reference from this PR [Tests](https://github.com/Expensify/App/pull/34716#issue-2087909126)

<details>
   <summary>BankAccountList Mock data</summary>
   
   ```javascript
Onyx.merge('bankAccountList', {
    1: {
        title: `Generic bank Checking`,
        description: 'Account ending in 5087',
        methodID: '1',
        key: `bankAccount-1`,
        accountType: 'bankAccount',
        accountData: {
            accountNumber: 'XXXXX5087',
            additionalData: {
                bankName: `Generic bank`,
            },
            state: 'OPEN',
            type: 'PERSONAL',
        },
        isDefault: false,
    },
    2: {
        title: `Generic bank Checking 2`,
        description: 'Account ending in 5123',
        methodID: '2',
        key: `bankAccount-2`,
        accountType: 'bankAccount',
        accountData: {
            accountNumber: 'XXXXX5088',
            additionalData: {
                bankName: `Generic bank`,
            },
            state: 'OPEN',
            type: 'BUSINESS',
        },
        isDefault: false,
    },
});

Onyx.merge('fundList', {
    4: {
        title: 'Test card',
        description: 'Card ending in 4242',
        methodID: 4,
        key: 'card-4',
        accountType: 'debitCard',
        accountData: {
            additionalData: {
                isBillingCard: false,
                isP2PDebitCard: true,
            },
            addressName: 'Test cardList card',
            addressState: 'US',
            addressStreet: 'Street',
            addressZip: 75092,
            cardMonth: 12,
            cardNumber: '424242XXXXXX4242',
            cardYear: 2026,
            created: '2023-01-13 22:27:23',
            currency: 'USD',
            fundID: 4,
        },
        isDefault: false,
    },
    5: {
        title: 'Test card 2',
        description: 'Card ending in 0000',
        methodID: 5,
        key: 'card-5',
        accountType: 'debitCard',
        accountData: {
            additionalData: {
                isBillingCard: false,
                isP2PDebitCard: true,
            },
            addressName: 'Nowhere',
            addressState: 'US',
            addressStreet: 'Street',
            addressZip: 75092,
            cardMonth: 12,
            cardNumber: '424242XXXXXX0000',
            cardYear: 2026,
            created: '2023-01-13 22:27:23',
            currency: 'USD',
            fundID: 5,
        },
        isDefault: false,
    },
});
```
</details>

<details>
<summary>Wallet mock data</summary>

```javascript

Onyx.merge('userWallet', {
    currentBalance: 1000,
    tierName: 'GOLD',
})

```

</details>

<details>
<summary>Result</summary>


https://github.com/Expensify/App/assets/35371050/f2ecff11-dcd9-4d96-b5d9-37790832fecc



</details>

**Comment by bernhardoj** [comment]
@jayeshmangwani thanks. @kavimuru can you please try the steps above?

**Comment by mountiny** [comment]
@shawnborton has kindly QAed this and works well! 🎉

**Pull request merged**